stubs/pyinstaller/PyInstaller/depend/analysis.pyi

```diff
@@ -2,17 +2,25 @@
 
 # The documentation explicitely mentions that "Normally you do not need to know about the module-graph."
 # However, some PyiModuleGraph typed class attributes are still documented as existing in imphookapi.
-from _typeshed import Incomplete
+from _typeshed import Incomplete, StrOrBytesPath, SupportsKeysAndGetItem
+from collections.abc import Iterable
+from typing_extensions import TypeAlias
+
+from PyInstaller.lib.modulegraph.modulegraph import Alias, Node
+
+_LazyNode: TypeAlias = Iterable[Node] | Iterable[str] | Alias | None
+# from altgraph.Graph import Graph
+_Graph: TypeAlias = Incomplete
 
 class PyiModuleGraph:  # incomplete
     def __init__(
         self,
         pyi_homepath: str,
-        user_hook_dirs=...,
-        excludes=...,
-        path: Incomplete | None = ...,
-        replace_paths=...,
-        implies=...,
-        graph: Incomplete | None = ...,
+        user_hook_dirs: Iterable[StrOrBytesPath] = ...,
+        excludes: Iterable[str] = ...,
+        path: Iterable[str] | None = ...,
```

Comment:
These and further args are keyword-only, since they're forwarded through `**kwargs`.

Reply:
Didn't know [keyword-only arguments](https://peps.python.org/pep-3102/) were a thing. Thanks!